src/main/scala/scala/tools/refactoring/implementations/OrganizeImports.scala

```diff
@@ -464,8 +463,82 @@ abstract class OrganizeImports extends MultiStageRefactoring with TreeFactory wi
       case (p, existingImports, others) =>
         val imports = scala.Function.chain(participants)(existingImports)
         p copy (stats = imports ::: others) replaces p
+    } &> transformation[Tree, Tree] {
+      case p: PackageDef =>
+        InnerImports.organizeImportsInMethodBlocks(p).replaces(p)
     }
 
     Right(transformFile(selection.file, organizeImports |> topdown(matchingChildren(organizeImports))))
   }
+
+  object InnerImports {
+    class RemoveUnused(block: Tree) extends Participant {
+      private def asText(t: Tree) = createText(stripPositions(t))
+
+      private def treeWithoutImports(tree: Tree) = new Transformer {
+        override def transform(tree: Tree): Tree = tree match {
+          case Import(_, _) => EmptyTree
+          case t => super.transform(t)
+        }
+      }.transform(tree)
+
+      private def allSelects(block: Tree) = {
+        val selectsTraverser = new Traverser {
+          import scala.collection.mutable
+          val selects = mutable.ListBuffer[Select]()
+          override def traverse(tree: Tree): Unit = tree match {
+            case s @ Select(qual, _) =>
+              selects += s
+              traverse(qual)
+            case t => super.traverse(t)
+          }
+        }
+        selectsTraverser.traverse(treeWithoutImports(block))
+        selectsTraverser.selects.map { asText }.toList
+      }
```

Comment:
As it turned out, the `allSelects` method implementation is problematic. You are trying to print the tree, which leads to an exception (see ticket [#1002476](https://scala-ide-portfolio.assembla.com/spaces/scala-ide/tickets/1002476)). Test case:

```scala
  @Test
  def xxx() = new FileSet {
    """
    /*<-*/
    package test

    object X {
      val m = Map[String, String]()

      def f = {
        for ((a, b) ← m)
          println((a,b))
        0
      }
    }
    """ becomes {
    """
    /*<-*/
    package test

    object X {
      val m = Map[String, String]()

      def f = {
        for ((a, b) ← m)
          println((a,b))
        0
      }
    }
    """
    }
  } applyRefactoring organizeCustomized(dependencies = Dependencies.RecomputeAndModify)
```

The problem is that the for comprehension desugars into a lot of synthetic code and when we try to print it it breaks. In general we should avoid tree printing whenever possible. I'm going to investigate how to resolve the issue in this case.